packages/web/components/place-limit-tool/index.tsx

```diff
@@ -0,0 +1,185 @@
+import { WalletStatus } from "@cosmos-kit/core";
+import { Dec } from "@keplr-wallet/unit";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import { FunctionComponent, useMemo, useState } from "react";
+
+import { Icon } from "~/components/assets";
+import { TokenSelectLimit } from "~/components/control/token-select-limit";
+import { LimitInput } from "~/components/input/limit-input";
+import { Tooltip } from "~/components/tooltip";
+import { Button } from "~/components/ui/button";
+import { useTranslation } from "~/hooks";
+import { OrderDirection, usePlaceLimit } from "~/hooks/limit-orders";
+import { useOrderbookPool } from "~/hooks/limit-orders/use-orderbook-pool";
+import { useStore } from "~/stores";
+import { formatPretty } from "~/utils/formatter";
+
+export interface PlaceLimitToolProps {
+  orderDirection: OrderDirection;
+}
+
+const percentAdjustmentOptions = [
+  { value: new Dec(0), label: "0%" },
+  { value: new Dec(0.02), label: "2%" },
+  { value: new Dec(0.05), label: "5%" },
+  { value: new Dec(0.1), label: "10%" },
+];
+
+export const PlaceLimitTool: FunctionComponent<PlaceLimitToolProps> = observer(
+  ({ orderDirection = OrderDirection.Bid }) => {
+    const { accountStore } = useStore();
+    const { t } = useTranslation();
+    const [baseDenom, setBaseDenom] = useState<string>("ION");
+    const quoteDenom = "OSMO";
+
+    const { poolId, orderbookContractAddress } = useOrderbookPool({
+      baseDenom,
+      quoteDenom,
+    });
+    const swapState = usePlaceLimit({
+      osmosisChainId: "localosmosis",
+      poolId,
+      orderDirection,
+      useQueryParams: false,
+      orderbookContractAddress,
+      baseDenom,
+      quoteDenom,
+    });
+    const account = accountStore.getWallet("localosmosis");
```

Comment:
Should we pass the osmosis chain ID? Or is this temporary?

We have an `accountStore.osmosisChainId` member to make this easy

Reply:
Temporary but will update it to use the `accountStore` reference 👍

Reply:
https://github.com/osmosis-labs/osmosis-frontend/pull/3308/commits/c8466446eaec7a82b02e6f4bbc2c2b614c1901b8